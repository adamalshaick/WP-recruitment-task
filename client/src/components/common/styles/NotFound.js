import styled from "styled-components";
import { Link } from "react-router-dom";

export const NotFoundWrapper = styled.section`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
`;

export const Button = styled(Link)`
  color: black;
  text-decoration: none;
  background-color: whitesmoke;
  padding: 1rem;
  border-radius: 5px;
  box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.25);
  margin: 1rem;
`;
