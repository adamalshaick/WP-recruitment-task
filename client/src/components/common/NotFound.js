import React from "react";
import { NotFoundWrapper, Button } from "./styles/NotFound";

export const NotFound = () => {
  return (
    <NotFoundWrapper>
      <h1>Sorry, this page doesn't exist</h1>
      <Button to="/">Back to the main page</Button>
    </NotFoundWrapper>
  );
};

export default NotFound;
