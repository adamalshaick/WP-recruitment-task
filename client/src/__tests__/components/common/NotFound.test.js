import React from "react";
import NotFound from "../../../components/common/NotFound";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const wrapper = shallow(<NotFound />);
describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
