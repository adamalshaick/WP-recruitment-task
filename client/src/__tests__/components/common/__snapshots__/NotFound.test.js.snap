// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<styled.section>
  <h1>
    Sorry, this page doesn't exist
  </h1>
  <Styled(Link)
    to="/"
  >
    Back to the main page
  </Styled(Link)>
</styled.section>
`;
